app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php

```diff
@@ -29,24 +29,29 @@ class ReportIncidentCommandHandler
      */
     public function handle(ReportIncidentCommand $command)
     {
-        $incident = Incident::create([
-            'name'         => $command->name,
-            'status'       => $command->status,
-            'message'      => $command->message,
-            'visible'      => $command->visible,
-            'component_id' => $command->component_id,
-        ]);
+        $data = [
+            'name'    => $command->name,
+            'status'  => $command->status,
+            'message' => $command->message,
+            'visible' => $command->visible,
+        ];
+
+        // Link with the component.
+        if ($command->component_id) {
+            $data['component_id'] = $command->component_id;
+        }
 
         // The incident occurred at a different time.
         if ($command->incident_date) {
             $incidentDate = Date::createFromFormat('d/m/Y H:i', $command->incident_date, config('cachet.timezone'))->setTimezone(Config::get('app.timezone'));
```

Comment:
we need to bind the handler to the ioc and inject the config value at that point